src/api/localStorage.js

```diff
@@ -0,0 +1,96 @@
+export class LocalStorage {
+  constructor() {
+    this.users = JSON.parse(localStorage.getItem('UserAccounts')) || [];
+  }
+
+  createUser(user) {
+    this.updateUsers();
+    user.id = this.generateId();
+    this.users.push(user);
+    this.setLocalStorageData();
+  }
+
+  setLocalStorageData() {
+    localStorage.setItem('UserAccounts', JSON.stringify(this.users));
+  }
+
+  updateUsers() {
+    this.users = JSON.parse(localStorage.getItem('UserAccounts')) || [];
+  }
+
+  getUser(id) {
+    const user = this.users.find(user => String(user.id) === String(id));
+    return user;
+  }
+
+  getAllUsers() {
+    return this.users;
+  }
+
+  editUser(id, editName, editValue) {
+    try{
+      const editObj = {};
+      editObj[`${editName}`] = `${editValue}`;
+      const updatedUser = Object.assign(this.getUser(id), editObj);
```

Comment:
could be a bit simpler syntactically (maybe)
```suggestion
      const updatedUser = Object.assign(this.getUser(id), {editName: editValue});
```
and then you'd be able to remove the two lines above creating `editObj`

Reply:
Got this to work:      
const updatedUser = Object.assign(this.getUser(id), {[editName]: editValue});